### PR TITLE
feat(console): api resource permissions page

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- Refactor the API resource details page to support the permissions tab
- Add the permissions tab to the API resource details page.

### Todo
- Create Permission
- Delete Permission

## Testing

image

### CODE REVIEW

File: packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          render: ({ name }) => <div className={styles.name}>{name}</div>,
+        },
+        {
+          title: t('api_resources.description_column'),
+          dataIndex: 'description',
+          colSpan: 9,
+          render: ({ description }) => <div className={styles.description}>{description}</div>,
+        },
+        {
+          title: <div />,
```
Comment (simeng-li) [inline packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/index.tsx:46]: should leave null?

File: packages/console/src/pages/ApiResourceDetails/index.tsx [hunk 2]
```diff
@@ -46,43 +39,14 @@ const ApiResourceDetails = () => {
   const theme = useTheme();
   const Icon = theme === AppearanceMode.LightMode ? ApiResource : ApiResourceDark;
 
+  const isOnPermissionTab = pathname.endsWith(ApiResourceTabs.Permissions);
```
Comment (simeng-li) [inline packages/console/src/pages/ApiResourceDetails/index.tsx:42]: Should we use the route params to define those paths? And then using `useParams` to get the tab key. End with looks hacky.

File: packages/console/src/pages/ApiResourceDetails/index.module.scss [hunk 3]
```diff
@@ -1,5 +1,9 @@
 @use '@/scss/underscore' as _;
 
+.permissionPage {
+  height: 100%;
```
Comment (simeng-li) [inline packages/console/src/pages/ApiResourceDetails/index.module.scss:4]: min-height?

File: packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/index.tsx (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/index.tsx]: updated.

File: packages/console/src/pages/ApiResourceDetails/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApiResourceDetails/index.tsx]: updated.

File: packages/console/src/pages/ApiResourceDetails/index.module.scss (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApiResourceDetails/index.module.scss:4]: Pages with tables will scroll in the table, not on the page.
image

File: packages/console/src/pages/ApiResourceDetails/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/ApiResourceDetails/index.tsx]: updated with nested routes.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/console/src/pages/ApiResourceDetails/index.tsx [hunk 4]
```diff
@@ -156,41 +123,28 @@ const ApiResourceDetails = () => {
...
+              {
+                resource: data,
+                isDeleting,
+                isLogtoManagementApiResource,
+                onResourceUpdated: (resource: Resource) => {
+                  void mutate(resource);
+                },
+              } as ApiResourceDetailsOutletContext
+            }
+          />
```
Comment (charIeszhao) [inline packages/console/src/pages/ApiResourceDetails/index.tsx:145]: Why Outlet? I don't quite get it

Comment (xiaoyijun) [reply packages/console/src/pages/ApiResourceDetails/index.tsx:145]: Per offline discussion, use `Outlet` to handle tabs.

**Pull request merged**